### PR TITLE
[docs] [dagster-aws] add docs for PipesGlueClient

### PR DESCRIPTION
**Pull request by danielgafni**
## Summary & Motivation
Adding docs for Glue Dagster Pipes

### CODE REVIEW

**Comment by danielgafni** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22969?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#22969** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22969?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#22968** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22968?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#22967** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22967?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @danielgafni and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/content/concepts/dagster-pipes/aws-glue.mdx
Comment (erinkcochran87) [inline docs/content/concepts/dagster-pipes/aws-glue.mdx:84]:
```suggestion
---

## Step 4: Creating Dagster definitions

Next, add the `PipesGlueClient` resource to your project's <PyObject object="Definitions" /> object:
```

- Making this into its own section; title is just a suggestion
- Adding a `PyObject` for `Definitions`

File: docs/content/concepts/dagster-pipes/aws-glue.mdx
Comment (erinkcochran87) [inline docs/content/concepts/dagster-pipes/aws-glue.mdx:111]:
```suggestion
This will launch the AWS Glue job and monitor its status until it either fails or succeeds. A job failure will also cause the Dagster run to fail with an exception.
```

Fixing a few typos

File: docs/content/concepts/dagster-pipes/aws-glue.mdx
Comment (danielgafni) [reply docs/content/concepts/dagster-pipes/aws-glue.mdx:31]: I was assuming the user already has a working Glue job. In this case they would only need to insert these lines to get access to Pipes. 

But I agree we can just get to the full example right away

File: docs/content/concepts/dagster-pipes/aws-glue.mdx
Comment (erinkcochran87) [inline docs/content/concepts/dagster-pipes/aws-glue.mdx:73]:
```suggestion
---

## Step 3: Add the PipesGlueClient to Dagster code
```

My bad for missing this!

File: docs/content/concepts/dagster-pipes/aws-glue.mdx
Comment (erinkcochran87) [inline docs/content/concepts/dagster-pipes/aws-glue.mdx:102]:
```suggestion
## Step 4: Create Dagster definitions
```

Sorry if I did this, but lets do this instead so it's the same as the other headings

File: docs/content/concepts/dagster-pipes/aws-glue.mdx
Comment (erinkcochran87) [inline docs/content/concepts/dagster-pipes/aws-glue.mdx:135]:
```suggestion
## Related
```

Also sorry for not catching this earlier - we use `Related` for these sections

**Review by erinkcochran87 — APPROVED** [review]
Stamping to unblock you, but I found a few small tweaks I missed in the last round. Please address those before ya merge! 🚀

**Comment by danielgafni** [comment]
### Merge activity

* **Jul 17, 4:04 PM EDT**: @danielgafni started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/22969).

**Pull request merged**
